katportalclient/client.py

```diff
@@ -1421,6 +1345,249 @@ def sensors_histories(self, filters, start_time_sec, end_time_sec, include_value
+                'start_time': '2017-02-07 00:00:00',
+                'modified': '',
+                'content': 'katportalclient userlog creation content!',
+                'parent_id': '',
+                'user': {'email': 'cam@ska.ac.za', 'id': 1, 'name': 'CAM'},
+                'attachment_count': 0,
+                'id': 40,
+                'end_time': '2017-02-07 23:59:59'
+             }, {..}]
+        """
+        url = self.sitemap['userlogs'] + '/query?'
+        if start_time is None:
+            start_time = time.strftime('%Y-%m-%d 00:00:00')
+        if end_time is None:
+            end_time = time.strftime('%Y-%m-%d 23:59:59')
+        request_params = {
+            'start_time': start_time,
+            'end_time': end_time
+        }
+        query_string = urlencode(request_params)
+        response = yield self.authorized_fetch(
+            url='{}{}'.format(url, query_string), auth_token=self._session_id)
+        raise tornado.gen.Return(json.loads(response.body))
+
+    @tornado.gen.coroutine
+    def create_userlog(self, content, tag_ids=None, start_time=None,
+                       end_time=None):
+        """
+        Create a userlog with specified linked tags and content, start_time
+        and end_time.
+
+        Parameters
+        ----------
+        content: str
+            The content of the userlog, could be any text. Required.
+
+        tag_ids: list
+            A list of tag id's to link to this userlog. Optional.
+            Example: [1, 2, 3, ..]
+
+        start_time: str
+            A formatted datetime string used as the start time of the userlog.
+            Format: %Y-%m-%d %H:%m:%s. Optional.
+
+        end_time: str
+            A formatted datetime string used as the end time of the userlog.
+            Format: %Y-%m-%d %H:%m:%s. Optional.
+
+        Returns
+        -------
+        userlog: dict
+            The userlog that was created. Example:
+            {
+                'other_metadata': [],
+                'user_id': 1,
+                'attachments': [],
+                'tags': '[]',
+                'timestamp': '2017-02-07 08:47:22',
+                'start_time': '2017-02-07 00:00:00',
+                'modified': '',
+                'content': 'katportalclient userlog creation content!',
+                'parent_id': '',
+                'user': {'email': 'cam@ska.ac.za', 'id': 1, 'name': 'CAM'},
+                'attachment_count': 0,
+                'id': 40,
+                'end_time': '2017-02-07 23:59:59'
+             }
+        """
+        url = self.sitemap['userlogs']
+        new_userlog = {
+            'user': self._current_user_id,
+            'content': content
+        }
+        if start_time is not None:
+            new_userlog['start_time'] = start_time
+        if end_time is not None:
+            new_userlog['end_time'] = end_time
+        if tag_ids is not None:
+            new_userlog['tag_ids'] = tag_ids
+
```

Comment:
Not the content, no. The tag_ids are used to select tags from the database by id, so if this is not an array of ints, it fails to update/add.